onnxruntime/core/providers/coreml/builders/impl/binary_op_builder.cc

```diff
@@ -55,6 +56,44 @@ bool CheckIfBothInputShapesMatch(const Node& node, const logging::Logger& logger
 }
 }  // namespace
 
+// Add variadic inputs to the model builder
+// in onnx spec, some node allows variadic inputs, such as max(x, y, z, ...)
+// while in coreml, maximum op only allows two inputs maximum(x, y)
+// the conversion is doing the following:
+// max(x, y, z, ...) -> max(max(x, y), z, ...)
+#if defined(COREML_ENABLE_MLPROGRAM)
+static void AddVariadicInputsToMB(std::unique_ptr<CoreML::Specification::MILSpec::Operation>* op, ModelBuilder& model_builder, const Node& node,
+                                  const logging::Logger& logger) {
+  using namespace CoreML::Specification::MILSpec;
+  const auto& input_defs(node.InputDefs());
+  std::string_view layer_input_name_x = model_builder.GetUniqueName(node, "variadic");
+  auto input_dtype = input_defs[0]->TypeAsProto()->tensor_type().elem_type();
+  const int32_t elem_type = static_cast<int32_t>(input_dtype);
+  std::vector<int64_t> x0_shape, x1_shape;
+  GetShape(*input_defs[0], x0_shape, logger);
+  GetShape(*input_defs[1], x1_shape, logger);
+  if (x0_shape.size() < x1_shape.size()) {
+    std::swap(x0_shape, x1_shape);
+  }
+  std::fill(x0_shape.begin(), x0_shape.end(), -1);
```

Comment:
Can we get the rank of each input instead of the shape? 

Would be good to explain why we're adding a shape with -1 values in a comment so that's captured for the next developer that has to work on the code.

Reply:
If we only fetch rank and fill it with -1. We will lost all the known shapes.